File: Models/Interview.cs
using System.ComponentModel.DataAnnotations;

namespace JobFairPortal.Models
{
    public enum InterviewStatus
    {
        Queued,
        InProgress,
        Shortlisted,
        Hired,
        Rejected
    }

    public class Interview
    {
        [Key]
        public int InterviewId { get; set; }
        [Required]
        public int CompanyId { get; set; }
        [Required]
        public int StudentId { get; set; }
        [Required]
        public int? RequestId { get; set; }
        public InterviewStatus Status { get; set; } = InterviewStatus.Queued;
        public DateTime? ScheduledTime { get; set; }
        public int? DurationMinutes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public Company Company { get; set; } = null!;
        public Student Student { get; set; } = null!;
        public InterviewRequest? Request { get; set; }
    }

}

File: lib/model/interview.dart
class Interview {
  final int interviewId;
  final String companyName;
  final String? companyLogo;
  final DateTime? scheduledTime;
  final int? durationMinutes;
  final String room;
  final String status;

  Interview({
    required this.interviewId,
    required this.companyName,
    this.companyLogo,
    this.scheduledTime,
    this.durationMinutes,
    required this.room,
    required this.status,
  });

  factory Interview.fromJson(Map<String, dynamic> json) {
    return Interview(
      interviewId: json['interviewId'],
      companyName: json['companyName'],
      companyLogo: json['companyLogo'],
      scheduledTime: json['scheduledTime'] != null
          ? DateTime.parse(json['scheduledTime'])
          : null,
      durationMinutes: json['durationMinutes'],
      room: json['room'] ?? 'TBD',
      status: json['status'] ?? 'Queued',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'interviewId': interviewId,
      'companyName': companyName,
      'companyLogo': companyLogo,
      'scheduledTime': scheduledTime?.toIso8601String(),
      'durationMinutes': durationMinutes,
      'room': room,
      'status': status,
    };
  }
}
